core/src/main/java/com/linecorp/armeria/server/docs/MethodInfo.java

```diff
@@ -130,7 +152,6 @@ public MethodInfo(String serviceName, String name,
 
         this.returnTypeSignature = requireNonNull(returnTypeSignature, "returnTypeSignature");
         this.parameters = ImmutableList.copyOf(requireNonNull(parameters, "parameters"));
-        assert !useParameterAsRoot || this.parameters.size() == 1;
```

Comment:
I could tweak the assertion to look for a single root parameter but it seemed like overkill